lib/ansible/modules/windows/setup.ps1

```diff
@@ -199,6 +199,20 @@ if($gather_subset.Contains('distribution')) {
         ansible_os_family = "Windows"
         ansible_os_name = ($win32_os.Name.Split('|')[0]).Trim()
     }
+    switch ($win32_os.ProductType) {
```

Comment:
3 things
* The indenting and formatting are a bit off, you can easily simplify this with the below
* The fact should really be called `ansible_os_product_type` not `ansible_os_producttype`
* Move this before adding the keys above so that operation is only done once and all the `distribution` keys are added together. This makes it easier to see the grouping and maintainability in the future

```
$product_type = switch($win32_os.ProductType) {
    1 { "workstation" }
    2 { "domain_controller" }
    3 { "server" }
    default { "unknown" }
}
$ansible_facts += @{
    ansible_distribution = $win32_os.Caption
    ...  # the other keys here
    ansible_os_product_type = $product_type
}
```